extensions/control-plane/api/management-api/secrets-api/src/main/java/org/eclipse/edc/connector/api/management/secrets/SecretsApi.java

```diff
@@ -0,0 +1,125 @@
+                    @ApiResponse(responseCode = "400", description = "Request was malformed, e.g. id was null",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    @ApiResponse(responseCode = "404", description = "A secret with the given ID does not exist",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class))))
+            }
+    )
+    JsonObject getSecret(String id);
+
+    @Operation(description = "Removes a secret with the given ID if possible.",
+            responses = {
+                    @ApiResponse(responseCode = "204", description = "Secret was deleted successfully"),
+                    @ApiResponse(responseCode = "400", description = "Request was malformed, e.g. id was null",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    @ApiResponse(responseCode = "404", description = "A secret with the given ID does not exist",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class))))
+            })
+    void removeSecret(String id);
+
+    @Operation(description = "Updates a secret with the given ID if it exists. If the secret is not found, no further action is taken. ",
+            requestBody = @RequestBody(content = @Content(schema = @Schema(implementation = SecretInputSchema.class))),
+            responses = {
+                    @ApiResponse(responseCode = "204", description = "Secret was updated successfully"),
+                    @ApiResponse(responseCode = "400", description = "Request was malformed, e.g. id was null",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    @ApiResponse(responseCode = "404", description = "Secret could not be updated, because it does not exist.")
+            })
+    void updateSecret(JsonObject secret);
+
+    @Schema(name = "SecretInput", example = SecretInputSchema.SECRET_INPUT_EXAMPLE)
+    record SecretInputSchema(
+            @Schema(name = CONTEXT, requiredMode = REQUIRED)
+            Object context,
+            @Schema(name = ID)
+            String id,
+            @Schema(name = TYPE, example = EDC_SECRET_TYPE)
+            String type,
+            @Schema(name = EDC_SECRET_VALUE, requiredMode = REQUIRED)
+            String value
+    ) {
+        public static final String SECRET_INPUT_EXAMPLE = """
+                {
+                    "@context": { "@vocab": "https://w3id.org/edc/v0.0.1/ns/" },
+                    "@id": "secret-id",
+                    "value" : "secret-value"
```

Comment:
`@value` is a reserved JSON-LD keyword. Not sure if this could cause any troubles. Maybe @ndr_brt knows?

Reply:
it does not seem to be a problem, at least from what we have experimented

Reply:
I don't know XD maybe @wolf4ood 
it should be good anyway

Reply:
ok, then lets put it in, we can always fix it later should it be a problem.